core/src/database/migrations.cpp

```diff
@@ -442,5 +442,18 @@ namespace ledger {
         template <> void rollback<7>(soci::session& sql) {
             sql << "DROP TABLE bech32_parameters";
         }
+
+        template <> void migrate<8>(soci::session& sql) {
+            sql << "CREATE TABLE ripple_memos("
```

Comment:
When we will do select from this table, the memos for one transaction can appear in any order (sql give no garanties that select return records in the same order as it was inserted). But memos field is an array, that means the order of elemets is important, because binary representation of transaction will be different depends of the order of the element, and this is bad. I think we need to have index field here as well.

Reply:
[The documentation doesn’t state the order is important](https://developers.ripple.com/transaction-common-fields.html#memos-field). If we really, really want to order them, I think we could simply add primary key (it doesn’t have any right now).

Do we care about the order here?

Reply:
In the description of Array Field from https://developers.ripple.com/serialization.html#internal-format . `Arrays contain several object fields in their native binary format in a specific order.` Not sure exactly what does this mean, but IMO this is not good that binary serialization of Transaction from DB can be different, depending on how DB is implemented.

Reply:
Hm, yeah, that might be an issue then. So I guess I can either add an index field, or just a primary key?

Reply:
Maybe we can store binary representation of the whole Memo as one binary field. As we have already implement serialization/deserialization.

Reply:
Nah I think it’s overkill. I’ll go with one of your other solutions. :)